### PR TITLE
Swift func

### PR DESCRIPTION
**Pull request by EngOmarAhmedIOS**
### Type of Change

- Adding a new entry
- Editing an existing entry (fixing a typo, bug, issues, etc)
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/swift/concepts/functions/functions.md
Comment (SSwiniarski) [inline content/swift/concepts/functions/functions.md:3]:
```suggestion
Description: 'Functions are self-contained chunks of code that perform a specific task.'
```

File: content/swift/concepts/functions/functions.md
Comment (SSwiniarski) [inline content/swift/concepts/functions/functions.md:16]:
```suggestion
**Functions** are self-contained chunks of code that perform a specific task. A function is given a name that identifies what it does, and this name is used to “call” the function to perform its task when needed.
```
We avoid first- and second-person pronouns in Docs.

File: content/swift/concepts/functions/functions.md
Comment (SSwiniarski) [inline content/swift/concepts/functions/functions.md:18]:
```suggestion
## Defining and Calling Functions

```

File: content/swift/concepts/functions/functions.md
Comment (SSwiniarski) [inline content/swift/concepts/functions/functions.md:19]:
```suggestion
Any function has three important elements used to define it. First is the function name. This is used to call the function. Second is the number of parameters a function takes, parameters referring to the input the function takes when calling it. Lastly is the return type of the function, which refers to the output data type of the function. When defining the name of the function, the name should describe its functionality. 
```

File: content/swift/concepts/functions/functions.md
Comment (SSwiniarski) [inline content/swift/concepts/functions/functions.md:20]:
```suggestion

The figure below describes the syntax of functions in Swift:
```

File: content/swift/concepts/functions/functions.md
Comment (SSwiniarski) [inline content/swift/concepts/functions/functions.md:22]: Let's rename the image to `swift-functions-diagram.png`.

File: content/swift/concepts/functions/functions.md
Comment (SSwiniarski) [inline content/swift/concepts/functions/functions.md:30]:
```suggestion

For example, this function called `greet` takes a `String` parameter called `person`, and returns a `String` equal to "Hello" plus the `person` value. The `greet(person:)` function is called by passing it a `String` value after the `person:` argument label as shown below:
```

File: content/swift/concepts/functions/functions.md
Comment (SSwiniarski) [inline content/swift/concepts/functions/functions.md:37]:
```suggestion
## Multiple Parameters

```
Line break after header.

File: content/swift/concepts/functions/functions.md
Comment (SSwiniarski) [inline content/swift/concepts/functions/functions.md:51]:
```suggestion
## Returning Multiple Values

```
Line break after header.

File: content/swift/concepts/functions/functions.md
Comment (SSwiniarski) [inline content/swift/concepts/functions/functions.md:52]:
```suggestion
A tuple type can be used as the return type for a function to return multiple values as part of one compound return value. Each value within its parentheses needs to be labeled and assigned a type in the function definition.
```

File: content/swift/concepts/functions/functions.md
Comment (SSwiniarski) [inline content/swift/concepts/functions/functions.md:53]:
```suggestion

For example, this function takes array of `Int` as an input, and returns the smallest and largest number in the array.

```

File: content/swift/concepts/functions/functions.md
Comment (SSwiniarski) [inline content/swift/concepts/functions/functions.md:68]:
```suggestion

Because the tuple’s member values are named as part of the function’s return type, they can be accessed with dot syntax to retrieve the minimum and maximum found values, as done below:

```

File: content/swift/concepts/functions/functions.md
Comment (SSwiniarski) [inline content/swift/concepts/functions/functions.md:74]:
```suggestion

## Omitting Argument Labels

```
Line break before and after header.

File: content/swift/concepts/functions/functions.md
Comment (SSwiniarski) [inline content/swift/concepts/functions/functions.md:75]:
```suggestion
To avoid having an argument labeled, use the `_` instead of an explicit argument label for that parameter.

```

File: content/swift/concepts/functions/functions.md
Comment (SSwiniarski) [inline content/swift/concepts/functions/functions.md:85]:
```suggestion
## Functions With an Implicit Return

```

File: content/swift/concepts/functions/functions.md
Comment (SSwiniarski) [inline content/swift/concepts/functions/functions.md:86]:
```suggestion
If the whole function body consists of a single expression, the function implicitly returns the value of that expression. For example:

```

File: content/swift/concepts/functions/functions.md
Comment (SSwiniarski) [inline content/swift/concepts/functions/functions.md:93]:
```suggestion
## Default Parameters

```

File: content/swift/concepts/functions/functions.md
Comment (SSwiniarski) [inline content/swift/concepts/functions/functions.md:94]:
```suggestion
A default parameter has a value assigned to a parameter in the function’s definition. For example:

```

File: content/swift/concepts/functions/functions.md
Comment (SSwiniarski) [inline content/swift/concepts/functions/functions.md:104]:
```suggestion

When a function with a default parameter is called, an argument for that parameter is not required. If the argument is included, that value will overwrite the default value and be used in the function body.

```

File: content/swift/concepts/functions/functions.md
Comment (SSwiniarski) [inline content/swift/concepts/functions/functions.md:105]:
```suggestion
## Variadic Parameters

```

File: content/swift/concepts/functions/functions.md
Comment (SSwiniarski) [inline content/swift/concepts/functions/functions.md:106]:
```suggestion
A variadic parameter accepts zero or more values of a specified type. Variadic parameters are written by inserting three period characters (`...`) after the parameter’s type name. The example below calculates the arithmetic mean (also known as the average) for a list of numbers of any length:

```

File: content/swift/concepts/functions/functions.md
Comment (SSwiniarski) [inline content/swift/concepts/functions/functions.md:120]:
```suggestion

The values passed to a variadic parameter are made available within the function’s body as an array. For example, a variadic parameter with a name of `numbers` and a type of `Double...` is made available within the function’s body as a constant array called `numbers` of type `Double`.

```

File: content/swift/concepts/functions/functions.md
Comment (SSwiniarski) [inline content/swift/concepts/functions/functions.md:121]:
```suggestion
## In-Out Parameters

```

File: content/swift/concepts/functions/functions.md
Comment (SSwiniarski) [inline content/swift/concepts/functions/functions.md:122]:
```suggestion
To be able to change the value passed for a certain argument, an `inout` argument should be used. An in-out parameter is denoted by `inout` in the function definition, and when the function is called, its variable argument must be prepended with an `&`.

```

File: content/swift/concepts/functions/functions.md
Comment (SSwiniarski) [inline content/swift/concepts/functions/functions.md:123]:
```suggestion
> **Note**: in-out parameters can’t have default values, and variadic parameters can’t be marked as `inout`.

```

File: content/swift/concepts/functions/functions.md
Comment (SSwiniarski) [inline content/swift/concepts/functions/functions.md:131]:
```suggestion

This function swaps the value of a and b.

```

File: content/swift/concepts/functions/functions.md
Comment (SSwiniarski) [inline content/swift/concepts/functions/functions.md:139]:
```suggestion

The example above shows that the original values of `someInt` and `anotherInt` are modified by the `swapTwoInts(_:_:)` function, even though the two variables were originally defined outside of the function.

```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@EngOmarAhmedIOS I did a first review and left a number of comments. Note as you edit that there should be a blank line separating paragraphs, headers, and code blocks. Many of my edits insert these blank lines.

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@EngOmarAhmedIOS I did a first review and left a number of comments. Note as you edit that there should be a blank line separating paragraphs, headers, and code blocks. Many of my edits insert these blank lines.

**Review by SSwiniarski — APPROVED** [review]
@EngOmarAhmedIOS I did one small change and this is ready for a second review.

**Comment by EngOmarAhmedIOS** [comment]
it's been more than a week waiting for a second review, is there anything I need to do. @SSwiniarski

**Comment by SSwiniarski** [comment]
@EngOmarAhmedIOS No, you're fine, we just have a limited number of maintainers and a lot of PRs. Have patience.

File: content/swift/concepts/functions/functions.md
Comment (caupolicandiaz) [inline content/swift/concepts/functions/functions.md:13]:
```suggestion
  - 'paths/build-ios-apps-with-swiftui'
```

File: content/swift/concepts/functions/functions.md
Comment (caupolicandiaz) [inline content/swift/concepts/functions/functions.md:3]:
```suggestion
Description: 'Functions are a type of structure that allows a set of operations to be packaged and repeated as desired.'
```

Comment (caupolicandiaz) [reply content/swift/concepts/functions/functions.md:3]: I noticed that the original description was an exact copy of the language used in the swift documentation. That is considered plagiarism, and it is not permitted in docs. 

It appears that some of the examples, headers, and other content are also copied directly from the documentation. All content in docs must be original. I would advise that you update this entry to remove/replace all copied content.

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@EngOmarAhmedIOS I've added a couple of comments. Please review carefully and update the entry as requested. Tag me when you've finished making the necessary changes. Thanks

File: content/swift/concepts/functions/functions.md
Comment (EngOmarAhmedIOS) [reply content/swift/concepts/functions/functions.md:3]: I don't mind to change the description but the header will always be the same because this are the things you need to talk about when talking about functions in swift, for examples I will change them you are right about that. @caupolicandiaz

**Comment by EngOmarAhmedIOS** [comment]
I did all the changes you @caupolicandiaz  asked for, I hope you remove the invalid label 😅. Have a nice day.

File: content/swift/concepts/functions/functions.md
Comment (caupolicandiaz) [inline content/swift/concepts/functions/functions.md:61]: This example has not been updated and is a direct copy from the swift documentation.

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@EngOmarAhmedIOS I've added some additional comments. Please update the entry, and tag me when the changes are complete.

File: content/swift/concepts/functions/functions.md
Comment (EngOmarAhmedIOS) [reply content/swift/concepts/functions/functions.md:61]: I changed , I wrote the code myself to avoid having the same problem, it just solves the same problem. @caupolicandiaz

**Comment by EngOmarAhmedIOS** [comment]
I did all the changes you asked for @caupolicandiaz , Have a good day.

File: content/swift/concepts/functions/functions.md
Comment (caupolicandiaz) [reply content/swift/concepts/functions/functions.md:62]: I updated the second block to use an original variable name and a different array

**Review by caupolicandiaz — APPROVED** [review]
@EngOmarAhmedIOS @SSwiniarski I just made a few additional edits. Looks good to go

**Pull request merged**
